contracts/sysio.roa/src/sysio.roa.cpp

```diff
@@ -546,6 +547,80 @@ namespace sysio {
 
         return asset(allocation_amount, state.total_sys.symbol);
     };
+
+    name roa::newuser(const name& creator, const name& nonce, const public_key& pubkey) {
+        require_auth(creator);
+
+       roastate_t roastate(get_self(), get_self().value);
+       auto state = roastate.get();
+       check(state.is_active, "ROA is not active yet");
+
+       // Check if creator is a tier-1 node owner
+       nodeowners_t nodeowners(get_self(), state.network_gen);
+       auto node_itr = nodeowners.find(creator.value);
+       check(node_itr != nodeowners.end() && node_itr->tier == 1, "Creator is not a registered tier-1 node owner");
+
+        // Check if nonce already exists in sponsors table for this creator
+        sponsors_t sponsors(get_self(), creator.value);
+        auto sponsor_itr = sponsors.find(nonce.value);
+        check(sponsor_itr == sponsors.end(), "Sponsor entry for this nonce already exists");
```

Comment:
Why do you care? You provide three attempts for the hash below.
You could include the current size of the sponsor table into the hash below which provides a diff hash for each new entry.